src/libraries/System.Net.Http/src/System/Net/Http/Headers/NameValueHeaderValue.cs

```diff
@@ -354,11 +354,27 @@ private static void CheckNameValueFormat(string name, string value)
 
         private static void CheckValueFormat(string value)
         {
-            // Either value is null/empty or a valid token/quoted string
-            if (!(string.IsNullOrEmpty(value) || (GetValueLength(value, 0) == value.Length)))
+            // Either value is null/empty or a valid token/quoted string https://tools.ietf.org/html/rfc7230#section-3.2.6
+            if (string.IsNullOrEmpty(value))
+            {
+                return;
+            }
+
+            // Trailing/leading space are not allowed
+            if (value[0] == ' ' || value[^1] == ' ')
```

Comment:
Should we be checking for tabs here as well?

Reply:
It's not super clear but I think so from https://tools.ietf.org/html/rfc7230#section-3.2.4
```
   OWS            = *( SP / HTAB )
   ...
   The field value does not
   include any leading or trailing whitespace: OWS occurring before the
   first non-whitespace octet of the field value or after the last
   non-whitespace octet of the field value ought to be excluded by
   parsers when extracting the field value from a header field.
```
I'll exclude HTAB and we'll see if it was expected(if covered in tests)

Reply:
In past code throw for leading HTAB thanks to `HttpRuleParser.GetQuotedStringLength` but not for leading, so I added check for leading/trailing.

@jkotalik or @scalablecory can you please run outerloops?I cannot run CI yet.

Reply:
having some chats with our team right now to find out why you can't run outerloop still :/

Reply:
Thanks a lot Cory!